### PR TITLE
Drastically improve Menhir syntax highlighting

### PR DESCRIPTION
**Pull request by ALANVF**
Menhir's syntax highlighting was bad so I decided to fix it.

Possible future improvements:
1) Fix highlighting for attributes (not possible without `#2` below).
2) Generate grammars using OCaml because JSON is terrible.

A lengthy example:
image

Another example (new merlin syntax):
image

### CODE REVIEW

**Review by mnxn — APPROVED** [review]

File: syntaxes/menhir.json [hunk 1]
```diff
@@ -66,9 +66,21 @@
...
         {
-          "comment": "builtin declaration",
+          "comment": "parameter declaration",
           "name": "keyword.other.menhir",
-          "match": "%(?:parameter|attribute)\\b"
+          "match": "%parameter\\b"
+        },
+        {
+          "comment": "attribute declaration",
+          "begin": "%(?:attribute\\b)?",
```
Comment (ALANVF) [inline syntaxes/menhir.json:93]: This will break stuff

**Review by ALANVF — COMMENTED** [review]
Pls fix

File: syntaxes/menhir.json (hunk 1 above)
Comment (mnxn) [reply syntaxes/menhir.json:93]: Please be more descriptive.

Can you give an example?

Comment (ALANVF) [reply syntaxes/menhir.json:93]: This might eat up `%%` when matched, and in general the `?` seems to be unnecessary.

**Comment by mnxn** [comment]
Thank you for your contribution!

**Pull request merged**